src/sinks/kafka.rs

```diff
@@ -392,6 +405,28 @@ impl Sink<Event> for KafkaSink {
     }
 }
 
+fn get_headers(event: &Event, headers_key: &str) -> Option<OwnedHeaders> {
+    if let Event::Log(log) = event {
+        if let Some(headers) = log.get(headers_key) {
+            match headers {
+                Value::Map(headers_map) => {
+                    let mut owned_headers = OwnedHeaders::new_with_capacity(headers_map.len());
+                    for (key, value) in headers_map {
+                        if let Value::Bytes(value_bytes) = value {
+                            owned_headers = owned_headers.add(key, value_bytes.as_ref());
+                        } else {
+                            warn!("Header value for key {} must be a UTF-8 encoded String (Bytes). Ignoring.", key);
+                        }
+                    }
+                    return Some(owned_headers);
+                }
+                _ => warn!("Value of headers_key field must be a map. Ignoring."),
+            }
+        }
+    }
+    None
```

Comment:
I so want this to be a `match`, but the logic ends up with a lot of `None` branches, so this is the simpler path.